### PR TITLE
Yes no dialog

### PR DESCRIPTION
**Pull request by saurabhB-Webonise**

### CODE REVIEW

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/YesNoVfDialogFragment.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.MutableLiveData
+import androidx.navigation.findNavController
+import com.sjana.fitness.R
+import com.sjana.fitness.base.BaseDialogFragment
+import com.sjana.fitness.view.feature.home.FragmentCommunicatorModel
+import kotlinx.android.synthetic.main.yesno_vf_dialog_fragment.*
+import javax.inject.Inject
+
+class YesNoVfDialogFragment : BaseDialogFragment<ValueFinderViewModel>() {
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/YesNoVfDialogFragment.kt:16]: No need of this class. This dialog is already present use that.

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/TopTenSelectedValueFragment.kt [hunk 2]
```diff
@@ -35,16 +38,31 @@ class TopTenSelectedValueFragment : BaseFragment<ValueFinderViewModel>() {
...
         })
         beginVfBtn.setOnClickListener({
             val action = TopTenSelectedValueFragmentDirections.topTenSelectedValueTovalueFinder(5,
                     getUnCheckedList(args.valueCarddataList.toMutableList()))
             it.findNavController().navigate(action)
         })
+
+        communicatorModel.dialogEvent.observe(viewLifecycleOwner, Observer<Boolean> {
+            if (it) {
+                findNavController().navigateUp()
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/TopTenSelectedValueFragment.kt:51]: what is this logic for? two times navigate up. this is incorrect fix it.

File: app/src/main/res/layout/yesno_vf_dialog_fragment.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.google.android.material.card.MaterialCardView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/yesno_vf_dialog_fragment.xml:2]: remove this file as well

**Review by tejasAlsi-Webonise — CHANGES_REQUESTED** [review]
Check comments

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/YesNoVfDialogFragment.kt (hunk 1 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/YesNoVfDialogFragment.kt]: done

File: app/src/main/res/layout/yesno_vf_dialog_fragment.xml (hunk 3 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/yesno_vf_dialog_fragment.xml]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/TopTenSelectedValueFragment.kt (hunk 2 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/TopTenSelectedValueFragment.kt]: done

**Review by tejasAlsi-Webonise — APPROVED** [review]

**Pull request merged**